cpp/include/raft/linalg/sqrt.cuh

```diff
@@ -34,14 +35,54 @@ namespace linalg {
  * @param stream cuda stream where to launch work
  * @{
  */
-template <typename math_t, typename IdxType = int>
-void sqrt(math_t* out, const math_t* in, IdxType len, cudaStream_t stream)
+template <typename in_t, typename out_t = in_t, typename IdxType = int>
+void sqrt(out_t* out, const in_t* in, IdxType len, cudaStream_t stream)
 {
   raft::linalg::unaryOp(
-    out, in, len, [] __device__(math_t in) { return raft::mySqrt(in); }, stream);
+    out, in, len, [] __device__(in_t in) { return raft::mySqrt(in); }, stream);
 }
 /** @} */
 
+/**
+ * @defgroup power Sqrt Arithmetic
+ * @{
+ */
+
+/**
+ * @brief Elementwise sqrt operation on the input buffer
```

Comment:
maybe input buffer isn't the best word to use here now that it's working on mdspans?